src/puppetlabs/puppetdb/http/server.clj

```diff
@@ -30,9 +42,30 @@
     (format "The %s API has been retired; please use v4" version)
     404)))
 
-(def routes
-  (app
-   ["v4" &] {:any v4-app}
+(defn query-app
```

Comment:
nothing specifically wrong, but it's nice to be able to see at a glance what endpoints are supported on what AAPI versions. Changing it this way would mean you would have to dig separately into each http namespace to see that information (although if we combine those namespaces the way we discussed today maybe that concern goes away.) Not sure that I have a suggestion that would be less clunky than what we had before. I think what you have is alright.